### Question 1

Run docker with the python:3.13 image. Use an entrypoint bash to interact with the container.

What's the version of pip in the image?

In [2]:
!docker run -it --entrypoint bash python:3.13

Unable to find image 'python:3.13' locally
3.13: Pulling from library/python

010cab7f: Pulling fs layer 
5b6b6766: Pulling fs layer 
29762372: Pulling fs layer 
eb567ed4: Pulling fs layer 
fab23101: Pulling fs layer 
20e3d16b: Pulling fs layer 
Digest: sha256:c8b03b4e98b39cfb180a5ea13ae5ee39039a8f75ccf52fe6d5c216eed6e1be1d[6A
Status: Downloaded newer image for python:3.13
root@7ca7f3127a33:/# ^C

root@7ca7f3127a33:/# 

In [3]:
!docker run --rm python:3.13 pip --version

pip 25.3 from /usr/local/lib/python3.13/site-packages/pip (python 3.13)


Answer: 25.3

### Question 2

services:
  db:
    container_name: postgres
    image: postgres:17-alpine
    environment:
      POSTGRES_USER: 'postgres'
      POSTGRES_PASSWORD: 'postgres'
      POSTGRES_DB: 'ny_taxi'
    ports:
      - '5433:5432'
    volumes:
      - vol-pgdata:/var/lib/postgresql/data

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4:latest
    environment:
      PGADMIN_DEFAULT_EMAIL: "pgadmin@pgadmin.com"
      PGADMIN_DEFAULT_PASSWORD: "pgadmin"
    ports:
      - "8080:80"
    volumes:
      - vol-pgadmin_data:/var/lib/pgadmin

volumes:
  vol-pgdata:
    name: vol-pgdata
  vol-pgadmin_data:
    name: vol-pgadmin_data